astropy/time/formats.py

```diff
@@ -664,6 +664,77 @@ def to_value(self, timezone=None, parent=None):
     value = property(to_value)
 
 
+class TimeYMDHMS(TimeUnique):
+    """
+    ymdhms: A Time format to represent Time as year, month, day, hour,
+    minute, second (thus the name ymdhms)
+    Example::
+
+      >>> from astropy.time import Time
+      >>> t = Time({'year': [2015, 2016], 'minute': [1, 2]}, scale='utc')
+      >>> t.iso
+      array(['2015-01-01 00:01:00.000', '2016-01-01 00:02:00.000'], dtype='<U23')
+      >>> t.ymdhms['year']
+      array([2015, 2016], dtype=int32)
+    """
+    name = 'ymdhms'
+
+    def _check_val_type(self, val1, val2):
+        # Note: don't care about val2 for this class
+        if not all(isinstance(val, dict) or
+                   (isinstance(val, np.record) and val.dtype.kind == 'V')
+                   for val in val1.flat):
+            raise TypeError('Input values for {0} class must be '
+                            'dict or numpy recarray or numpy structured'
+                            ' array objects'.format(self.name))
+        return val1, None
+
+    def set_jds(self, val1, val2):
+        if val1.dtype.kind == 'V':
+            times_dict = dict([(dtype, val1[dtype])
+                               for dtype in val1.dtype.names])
+        else:
+            times_dict = val1.item()
+
+        jd1, jd2 = erfa.dtf2d(self.scale.upper().encode('ascii'),
+                              times_dict['year'],
+                              times_dict.get('month', 1),
+                              times_dict.get('day', 1),
+                              times_dict.get('hour', 0),
+                              times_dict.get('minute', 0),
+                              times_dict.get('second', 0))
+        self.jd1, self.jd2 = day_frac(jd1, jd2)
+
+    @property
+    def value(self):
+        scale = self.scale.upper().encode('ascii')
+        iys, ims, ids, ihmsfs = erfa.d2dtf(scale, self.precision,
+                                           self.jd1, self.jd2_filled)
+        ihrs = ihmsfs['h']
+        imins = ihmsfs['m']
+        isecs = ihmsfs['s']
+        ifracs = ihmsfs['f']
+
+        shape = 1 if not self.jd1.shape else self.jd1.shape
+        out = np.empty(shape, dtype=[
+                                ('year', 'i4'),
+                                ('month', 'i4'),
+                                ('day', 'i4'),
+                                ('hour', 'i4'),
+                                ('minute', 'i4'),
+                                ('second', 'f8'),
+                        ])
+        out['year'] = iys
+        out['month'] = ims
+        out['day'] = ids
+        out['hour'] = ihrs
+        out['minute'] = imins
+        out['second'] = isecs + ifracs * 10**(-self.precision)
+        out = out.view(np.recarray)
```

Comment:
Why cast as `np.recarray`?  I generally prefer pure np structured array.

Reply:
Because @mhvk said so. Also I kind of prefer doing `Time.ymdhms.year` rather than `Time.ymdhms['year']`.

If we are using recarray, it would (partially, at least) close #3831.

Reply:
@vn-ki - fair enough, recarray makes sense.  But then the scalar case definitely needs to be a zero-dim recarray (not dict) to have consistent API.